source/blog/2016-06-14-rspec-3.5-has-been-released.md

```diff
@@ -0,0 +1,125 @@
+---
+title: RSpec 3.5 has been released!
+author: Sam Phippen
+---
+
+RSpec 3.5 has just been released! Given our commitment to
+[semantic versioning](http://semver.org/), this should be a trivial
+upgrade for anyone already using any RSpec 3 release, but if we did introduce
+any regressions, please let us know, and we'll get a patch release
+out with a fix ASAP.
+
+RSpec continues to be a community-driven project with contributors
+from all over the world. This release includes over 450 commits and 150
+merged pull requests from over 50 different contributors!
+
+Thank you to everyone who helped make this release happen!
+
+## Notable Changes
+
+### Rails: Support for Rails 5
+
+The headline here is that RSpec 3.5.0 is compatible with Rails 5. As Rails 5
+betas and release candidates have been released, we've been releasing betas of
+3.5.0 to keep up alongside Rails. Due to this being a major release of Rails,
+but not a major release of RSpec, Rails has removed some APIs that we consume.
+This only gets exposed to you, our users, in one place: controller testing.
+
+In [Rails 5](https://github.com/rails/rails/issues/18950) `assigns` and
+`assert_template` are "soft deprecated". Controller tests themselves *are not*,
+and adding `:type => :controller` to your specs is still 100% supported.
+Through Rails 3 and 4: it was both prevalent and idiomatic to use `assigns` in
+controller specs. As this is a minor release of RSpec our commitment to SemVer
+means that we are not going to break your existing controller specs. For
+existing Rails applications that make heavy use of `assigns` adding the
+[`rails-controller-testing`](https://github.com/rails/rails-controller-testing)
+to one's Gemfile will restore `assigns` and `assert_template`. RSpec integrates
+with this gem seamlessly, so your controller specs should just continue to work.
+
+For new Rails apps: we don't recommend adding the `rails-controller-testing gem
+to your application. The "official" stance of the Rails team and the RSpec core
+team is that [request specs](https://www.relishapp.com/rspec/rspec-rails/docs/request-specs/request-spec)
+should now be written instead. Request specs allow one to focus on a single
+controller action, but unlike controller tests involve the router, the
+middleware stack, and both rack requests and responses. In Rails 5, they are
+very fast thanks to the work by [Eileen Uchitelle](https://twitter.com/eileencodes?lang=en-gb)
+<sup>[[1]](#footnotes)</a></sup> of the Rails Committer Team.
+
+The other important feature of Rails 5 we wanted to discuss is ActionCable.
+Unfortunately RSpec is not able to provide a simple testing story for
+ActionCable at this time. Rails is working on a testing type for ActionCable
+slated for release as part of Rails 5.1. We'll be watching that closely and work
+something up when it's ready. In the mean time, we suggest you test ActionCable
+through a browser, in an integrated fashion.
```

Comment:
Rails 5 has not support for automated tests against ActionCable?  Wow!  Weird.

Anyhow, instead of telling readers to just test manually (that seems awful), maybe we can tell them they'll need to test it in integrated fashion using a real web socket client such as https://github.com/shokai/websocket-client-simple ?  After all, there's no reason they couldn't write tests that boot rails and use a client like that to interact with it, right?